rotkehlchen/exchanges/coinbase.py

```diff
@@ -905,37 +907,58 @@ def _process_advanced_trade(self, event: dict[str, Any], timestamp: Timestamp) -
-                if order_id in self.advanced_trade_orders:
-                    log.debug('Ignoring already seen coinbase advanced trade other side', other_side=event)  # noqa: E501
-                    return None
-
-                self.advanced_trade_orders.add(order_id)
-
         else:
             trade_type = TradeType.BUY if amount < 0 else TradeType.SELL
 
+        order_id = event['advanced_trade_fill']['order_id']
+        if (
+            (existing_order_data := self.advanced_trade_orders.get(order_id)) is not None and
+            existing_order_data[0].trade_type != trade_type
+        ):
+            log.debug('Ignoring other side of already seen coinbase advanced trade', other_side=event)  # noqa: E501
+            return None
+
         # The base/quote asset is determined from the product id
         base_asset = asset_from_coinbase(asset_identifiers[0], time=timestamp)
         quote_asset = asset_from_coinbase(asset_identifiers[1], time=timestamp)
         rate = Price(deserialize_fval(event['advanced_trade_fill']['fill_price'], name='fill_price', location='Coinbase advanced trade'))  # noqa: E501
         if tx_asset == quote_asset:  # calculate trade amount depending on the denominated asset
             amount /= rate  # type: ignore[assignment]  # mixing asset amount and fval types
 
+        amount = AssetAmount(abs(amount))
         fee_amount = abs(deserialize_fee(event['advanced_trade_fill']['commission']))
         fee_asset = quote_asset  # fee is always in quote asset
-        return Trade(
-            timestamp=timestamp,
-            location=Location.COINBASE,
-            base_asset=base_asset,
-            quote_asset=quote_asset,
-            trade_type=trade_type,
-            amount=AssetAmount(abs(amount)),
-            rate=rate,
-            fee=fee_amount,  # type: ignore  # abs() doesn't propagate Fee type
-            fee_currency=fee_asset,
-            link=str(event['id']),
-        )
+
+        if existing_order_data is None:  # first event for this order
+            self.advanced_trade_orders[order_id] = (
+                (new_trade := Trade(
+                    timestamp=timestamp,
+                    location=Location.COINBASE,
+                    base_asset=base_asset,
+                    quote_asset=quote_asset,
+                    trade_type=trade_type,
+                    amount=amount,
+                    rate=rate,
+                    fee=fee_amount,  # type: ignore  # abs() doesn't propagate Fee type
+                    fee_currency=fee_asset,
+                    link=str(event['id']),
+                )),
+                [(amount, rate)],
+            )
+            return new_trade
+        else:  # there are multiple events for this order - update the existing trade.
+            existing_trade = existing_order_data[0]
+            existing_trade.amount += amount  # type: ignore  # both are AssetAmount, but mypy sees it as FVal after adding them
```

Comment:
I believe they get added as FVal and not assetAmount. YOu would need to overwrite the __iadd__ method on assetAmount if I'm not mistaken